### PR TITLE
Fix storage tests

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/tests/test_retry_async.py [hunk 1]
```diff
@@ -47,6 +47,18 @@ class AiohttpTestTransport(AioHttpTransport):
             response.content_type = response.headers.get("content-type")
         return response
 
+class AiohttpRetryTestTransport(AioHttpTransport):
+    """Workaround to vcrpy bug: https://github.com/kevin1024/vcrpy/pull/461
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_retry_async.py]: Remove the comment, not the purpose of this

File: sdk/storage/azure-storage-blob/tests/test_retry.py [hunk 2]
```diff
@@ -32,6 +32,16 @@
...
 
+class RetryRequestTransport(RequestsTransport):
+    """Transport to test retry"""
+    def __init__(self, *args, **kwargs):
+        self.count = 0
+        super().__init__(*args, **kwargs)
+    
+    def send(self, request, **kwargs):
+        self.count += 1
+        response = super().send(request, **kwargs)
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_retry.py]: Unfortunately on Pyhon 2 tests, you need to long "super" declaration....

Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/test_retry.py]: pushed

File: sdk/storage/azure-storage-blob/tests/test_retry.py [hunk 3]
```diff
@@ -118,19 +128,20 @@ def test_retry_on_socket_timeout(self, resource_group, location, storage_account
...
         assert service._client._client._pipeline._transport.connection_config.timeout == 11
         assert service._client._client._pipeline._transport.connection_config.read_timeout == 0.000000000001
 
         # Act
         try:
-            with self.assertRaises(ServiceResponseError) as error:
+            with self.assertRaises(AzureError) as error:
                 service.create_container(container_name)
             # Assert
+            assert retry_transport.count == 4
```
Comment (mikeharder) [inline sdk/storage/azure-storage-blob/tests/test_retry.py]: Please add comment explaining why a retry count of 4 is expected.

**Pull request merged**